jetty9-base/src/main/java/com/google/cloud/runtimes/jetty9/TracingLogHandler.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright (C) 2016 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.runtimes.jetty9;
+
+import java.util.logging.ConsoleHandler;
+import java.util.logging.LogRecord;
+
+/**
+ * A Logging Handler extended with a request traceId label.
+ */
+public class TracingLogHandler extends ConsoleHandler {
+
+  /**
+   * Construct a TracingLogHandler.
+   */
+  public TracingLogHandler() {}
+
+  @Override
+  public synchronized void publish(LogRecord record) {
+    String traceId = RequestContextScope.getCurrentTraceId();
+    if (traceId != null) {
+      record.setMessage(record.getMessage() + " traceId=" + traceId);
```

Comment:
OK, I'm working on the formatter now.   Currently I'm basing it off the SimpleFormatter and just adding an extra positional field for traceId to the other 6 fields it already supports.   However, this requires a cut/paste of all the code, which does use a `LoggingSupport` class from `sun.util`, so I expect that is not OK and I'll have to avoid that?

Note that I add the TraceId to the parameters in the handler, but then remove it in the formatter, so the normal message formatting is not passed the extended parameter list.  This costs 2 array copies, which hopefully are not significant.